### DISCUSSION TITLE
Default value of quartz.threadPool.threadCount

### DISCUSSION THREAD

**Discussion by tejal2610** [quartznet/quartznet#1903] [Q&A]
I am using clustering in my project. There are multiple numbers of jobs running frequently. I have not mentioned quartz.threadPool.threadCount. So default value of quartz.threadPool.threadCount is -1 and what does that mean?

If I have multiple scheduled jobs should I specify threadCount other than using default?

**Comment by lahma** [comment]
The default should be 10, I don't know where you get that -1 from? You should tune the value based on the concurrency level you need per scheduler. Generally as they are being run as tasks .NET runtime should cope them with quite well without wasting resources when idle.

**Reply by tejal2610** [reply]
And if threadCount is more than 1 in clustered scheduler, also with added attribute **DisallowConcurrentExecution** will multiple threads can cause concurrent job with same jobKey?

**Reply by lahma** [reply]
> And if threadCount is more than 1 in clustered scheduler, also with added attribute **DisallowConcurrentExecution** will multiple threads can cause concurrent job with same jobKey?
> 

No.
